ha/core/event_analyzer/watcher/watcher.py

```diff
@@ -55,77 +51,60 @@ def __init__(self, consumer_id: int, message_type: str, consumer_group: str,
         self.parser = event_parser
         self.subscriber = subscriber
         self._validate()
-        self.consumer = self._get_connection()
+        self.consumer = MessageBus.get_consumer(consumer_id=str(self.consumer_id),
+                                consumer_group=self.consumer_group,
+                                message_type=self.message_type,
+                                callback=self.process_message)
 
     def _validate(self) -> None:
         """
         Validate watcher and raise exception.
-
         Raises:
             EventAnalyzer: event analyzer exception.
         """
         if not isinstance(self.subscriber, Subscriber):
             raise InvalidSubscriber(f"Invalid subscriber {self.subscriber}")
 
-    def _get_connection(self) -> MessageConsumer:
-        """
-        Get message consumer connection.
-
-        Returns:
-            MessageConsumer: Return instance of MessageConsumer.
-        """
-        return MessageConsumer(consumer_id=str(self.consumer_id),
-                                consumer_group=self.consumer_group,
-                                message_types=[self.message_type],
-                                auto_ack=False, offset='earliest')
-
-    def _get_message(self):
+    def process_message(self, message: str) -> None:
         """
-        Receive message from message bus.
-
-        Returns:
-            str: JSON object of message
+        Callback function to get message.
+        Args:
+            message (str): Message.
         """
         try:
-            message = self.consumer.receive(timeout=0)
+            message = json.loads(message.decode('utf-8'))
         except Exception as e:
-            Log.error(f"Failed to receive message, error: {e}. Retrying to receive.")
-            return None
+            Log.error(f"Invalid message {message}, sollow exception and ack it. Error: {e}")
+            return CONSUMER_STATUS.SUCCESS
+        if message is None:
+            return CONSUMER_STATUS.SUCCESS
```

Comment:
Here also, Message itself is None. So, after SUCCESS return, ack will be sent. How can an empty message will be acknowledged?

Reply:
it is not valid code 
```
        if message is None:
            return CONSUMER_STATUS.SUCCESS
```
will remove in next patch

Reply:
it was old code code will not reach here

How can an empty message will be acknowledged?
not sure but if message bus allows empty message also seen None in few message that time it is irrelevant message for watcher

as json.loads(None) will fail so no need to handle separately 
Removed

Reply:
Right. Got it.